interface/src/app/app.js

```diff
@@ -19,15 +20,37 @@ angular.module( 'adage', [
   $resourceProvider.defaults.stripTrailingSlashes = false;
 }])
 
-.run( function run () {
+.run( function run ( $state, $location ) {
+
+    // Will only redirect if there is no state AND the url is not a
+    // route for tribe_client
+    String.prototype.startsWith = function(prefix) {
+        return this.slice(0, prefix.length) == prefix;
+    };
+
+    if ($state['current']['name'] === '') {
+      if (window.location.pathname.startsWith('/tribe_client')) {
```

Comment:
Hmmm that seems like a possibility - another possibility that I was thinking of was just porting all of the interface functionality of tribe-client to our Angular app (so that we don't have to deal with those annoying urls). This way, the only way we are using the '/tribe_client/' urls is as API endpoints (to get the access token, get the user that is logged in, and the genesets), but we don't actually send the users to these urls in the browser.